### PR TITLE
[extractor/Pstream] Adding extractor

### LINKED ISSUE #4802: [Site Request] pstream.net
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.08.19** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

_No response_

### Example URLs

- Single video: https://www.pstream.net/e/mQN5YQKD6jLXvRa

### Provide a description that is worded well enough to be understood

- m3u8 and content not encrypted
- regex to get the m3u8 link : `https:\/\/www\.pstream\.net\/m\/([a-zA-Z0-9]*)\.m3u8\?expires=([0-9]*)&signature=(\b[A-Fa-f0-9]{64}\b)`
- m3u8 link example: `https://www.pstream.net/m/eyJpdiI6IlA1cjdmWjlHWmoxNWJOaG9CMzFmM3c9PSIsInZhbHVlIjoiQWRpZkEyRmJQck9UTFdNa0pSV2V3dz09IiwibWFjIjoiNTAzYTM1ZTUwZGEzNGU3ZTNiZTgzNzIxYjAwMDg4MmU3M2MzYjNjYzU5MjcyZDkxZDRkYTFjMzY0YTlhYzA0YyJ9.m3u8?expires=1661943140&signature=af55d305a56c48191188d3ff7406e41ab6fc37e78b4a71d97083e7b0c98b4cd6`

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.pstream.net/e/mQN5YQKD6jLXvRa']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.08.19.2 [48c88e088] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: bfbecd117
[debug] Python 3.10.4 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg N-107944-g4fce3bab64-20220830 (setts), ffprobe N-107944-g4fce3bab64-20220830
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1665 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.08.19, Current version: 2022.08.19.2
yt-dlp is up to date (2022.08.19.2)
[debug] [generic] Extracting URL: https://www.pstream.net/e/mQN5YQKD6jLXvRa
[generic] mQN5YQKD6jLXvRa: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] mQN5YQKD6jLXvRa: Extracting information
[debug] Looking for Brightcove embeds
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.pstream.net/e/mQN5YQKD6jLXvRa
Traceback (most recent call last):
  File "C:\Users\Romain\Documents\yt-dlp\yt_dlp\YoutubeDL.py", line 1459, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\Romain\Documents\yt-dlp\yt_dlp\YoutubeDL.py", line 1535, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\Romain\Documents\yt-dlp\yt_dlp\extractor\common.py", line 670, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\Romain\Documents\yt-dlp\yt_dlp\extractor\generic.py", line 3078, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.pstream.net/e/mQN5YQKD6jLXvRa
```

### PR DESCRIPTION
**Pull request by Starmania**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This adds support for pstream.net

Fixes #4802 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/pstream.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from .common import InfoExtractor
+from ..utils import base64
+
+
+class PStreamIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?pstream\.net/(e|v)/(?P<id>[a-zA-Z0-9]+)'
```
Comment (HobbyistDev) [inline yt_dlp/extractor/pstream.py:6]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?pstream\.net/(e|v)/(?P<id>\w+)'
```

File: yt_dlp/extractor/pstream.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            'thumbnail': None,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        url = f'https://www.pstream.net/e/{video_id}'
+
+        webpage = self._download_webpage(url, video_id)
```
Comment (HobbyistDev) [inline yt_dlp/extractor/pstream.py:33]: Inline `url` in `webpage`

File: yt_dlp/extractor/pstream.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        M3U8_RE = r'(https:\/\/www\.pstream\.net\/m\/(?:[a-zA-Z0-9]*)\.m3u8\?expires=(?:[0-9]*)&signature=(?:\b[A-Fa-f0-9]{64}\b))'
+
+        for i in goodJson.items():
+            try:
+                if self._search_regex(M3U8_RE, i[1], '0', fatal=False, default=None) is not None:
+                    m3u8_URL = i[1]
+            except TypeError:
+                pass
+
+        title = self._og_search_title(webpage)
```
Comment (HobbyistDev) [inline yt_dlp/extractor/pstream.py:58]: Inline in return

**Comment by HobbyistDev** [comment]
Does the test link still work for you ? i tried it using browser and curl, it give 404 error

**Comment by Starmania** [comment]
It is normal that you get an 404 error with curl, because you need to set the header `Accept-Language` to anything you want (Ex:`_`) and also you need to specify User agent

File: yt_dlp/extractor/pstream.py (hunk 2 above)
Comment (Starmania) [reply yt_dlp/extractor/pstream.py:33]: I think it's important to not inline it, when debugging, or just exploring the script, see that url is always changed by the embed version...

File: yt_dlp/extractor/pstream.py (hunk 3 above)
Comment (Starmania) [reply yt_dlp/extractor/pstream.py:58]: I won't inline it, have tested, too ugly... Keep my definitions just before return.

Comment (HobbyistDev) [reply yt_dlp/extractor/pstream.py:58]: Sorry, but did you read the contibution guideline ?  (ref: https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#inline-values)

Comment (Starmania) [reply yt_dlp/extractor/pstream.py:58]: ok, I give up, I respect the guidelines

**Comment by Starmania** [comment]
As you want, I will close it...

**Pull request closed without merging**